src/components/Exchange/PositionSeller.js

```diff
@@ -358,6 +361,14 @@ export default function PositionSeller(props) {
 
     receiveToken = isSwapAllowed && swapToToken ? swapToToken : collateralToken;
 
+    // receiveToken does not need to change for STOP order
+    if (isSwapAllowed && isContractAccount && isAddressZero(receiveToken.address)) {
+      const wrappedToken = toTokens.find((t) => t.baseSymbol === nativeTokenSymbol);
+      setSwapToToken(wrappedToken);
+      setSavedRecieveTokenAddress(wrappedToken.address);
```

Comment:
i think better to change state (call set* functions) inside e.g. some useEffect hook? otherwise by calling setState inside component render function you can in theory go into recursion and this is a common pattern to change state in handlers or hooks in general?

Reply:
Updated!